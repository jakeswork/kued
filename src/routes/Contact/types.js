import { shape } from 'prop-types';

export default {
  classes: shape({}),
};
