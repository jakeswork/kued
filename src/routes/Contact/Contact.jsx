import React, { Component } from 'react';

import propTypes from './types';
import MainLayout from '../../layouts/MainLayout';
import Text from '../../components/Text';
import Card from '../../components/Card';
import Emoji from '../../components/Emoji';
import ResponsiveModal from './components/ResponsiveModal';
import Button from '../../components/Button';

export const cardStyles = {
  cursor: 'pointer',
  minWidth: 240,
};

class Contact extends Component {
  static propTypes = propTypes;

  state = {
    showType: null,
  }

  close = cb => this.setState(({ showType: null }), typeof cb === 'function' ? cb : () => {})

  showHelp = showType => this.close(() => setTimeout(() => this.setState({ showType }), 300))

  render() {
    const { classes } = this.props;
    const { showType } = this.state;

    return (
      <MainLayout>
        <Text h1>
          How can we help?
        </Text>
        <Text caption>Click on a card to learn more.</Text>
        <div className={classes.cardsWrapper}>
          <Card
            style={cardStyles}
            onClick={() => this.showHelp('support')}
            data-test-id="supportCard"
          >
            <Text h4>
              Help us help you&nbsp;
              <Emoji icon="🤝" label="handshake" />
            </Text>
            <Text bold caption>Want to help out by supporting the project?</Text>
          </Card>
          <Card style={cardStyles} onClick={() => this.showHelp('bug')}>
            <Text h4>
              Something not right?&nbsp;
              <Emoji icon="🛠" label="hammer-and-wrench" />
            </Text>
            <Text bold caption>Let us know if you&apos;re having any problems.</Text>
          </Card>
          <Card style={cardStyles} onClick={() => this.showHelp('vote')}>
            <Text h4>
              Missing out?&nbsp;
              <Emoji icon="😭" label="loudly-crying-face" />
            </Text>
            <Text bold caption>Vote for us to come to your server.</Text>
          </Card>
        </div>
        <ResponsiveModal in={showType === 'support'} onClose={this.close}>
          <Text h3>Supporting Kued</Text>
          <Text>
            Kued is an independent project with ongoing development
            made possible thanks to the support of awesome backers.
            If you&apos;d like to help us out in the best way possible,
            please consider joining them!
          </Text>
          <Button>Donate</Button>
        </ResponsiveModal>
        <ResponsiveModal in={showType === 'bug'} onClose={this.close}>
          <Text h3>Reporting Issues</Text>
          <Text>
            We want to make Kued the best it can be. To do so, we need your help.
            There&apos;s only so much testing we can do. As such, we call on the users of Kued
            to report any problems they have so that we can tackle them head on
            and give all of our users the best experience.
          </Text>
          <Button>Report Problem</Button>
        </ResponsiveModal>
        <ResponsiveModal in={showType === 'vote'} onClose={this.close}>
          <Text h3>Server Requests</Text>
          <Text>
            Is Kued not yet supported by your server but you would like it to be?
            Let your voice be heard! Head over to the vote
            page to let us know where we&apos;re wanted!
          </Text>
          <Text bold caption>
            Once a server reaches 50 vote points it will be considered for development.
          </Text>
          <Button link="/vote">Vote</Button>
        </ResponsiveModal>
      </MainLayout>
    );
  }
}

export default Contact;
