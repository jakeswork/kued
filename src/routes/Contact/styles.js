export default {
  cardsWrapper: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
};
