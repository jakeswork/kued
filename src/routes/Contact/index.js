import withStyles from 'react-jss';

import Contact from './Contact';
import styles from './styles';

export default withStyles(styles)(Contact);
