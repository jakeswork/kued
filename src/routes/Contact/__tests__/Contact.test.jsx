import React from 'react';
import { shallow } from 'enzyme';

import Contact, { cardStyles } from '../Contact';

const defaultProps = {
  classes: {},
};

describe('The Contact instance', () => {
  describe('When called with default properties', () => {
    it('should render without throwing any errors', () => {
      const wrapper = shallow(<Contact {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when clicking on the cards', () => {
    it('should fire the showHelp method', () => {
      jest.useFakeTimers();

      const wrapper = shallow(<Contact {...defaultProps} />);
      const cards = wrapper.find({ style: cardStyles });

      cards.forEach((card, i) => {
        card.simulate('click');

        jest.advanceTimersByTime(300);

        if (i === 0) return expect(wrapper.state('showType')).toEqual('support');

        if (i === 1) return expect(wrapper.state('showType')).toEqual('bug');

        if (i === 2) return expect(wrapper.state('showType')).toEqual('vote');
      });
    });
  });

  describe('when calling the close method', () => {
    it('should set the state to null', () => {
      const wrapper = shallow(<Contact {...defaultProps} />);

      wrapper.setState({ showType: 'test' });

      expect(wrapper.state('showType')).toEqual('test');

      wrapper.instance().close();

      expect(wrapper.state('showType')).toEqual(null);
    });
  });
});
