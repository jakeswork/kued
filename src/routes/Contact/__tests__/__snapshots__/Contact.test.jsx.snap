// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Contact instance When called with default properties should render without throwing any errors 1`] = `
<Jss(MainLayout)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={true}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    style={Object {}}
  >
    How can we help?
  </Jss(Text)>
  <Jss(Text)
    bold={false}
    caption={true}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    style={Object {}}
  >
    Click on a card to learn more.
  </Jss(Text)>
  <div>
    <Jss(Card)
      data-test-id="supportCard"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
          "width": "25%",
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        style={Object {}}
      >
        Help us help you 
        <Jss(Emoji)
          emoji={null}
          icon="🤝"
          label="handshake"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        style={Object {}}
      >
        Want to help out by supporting the project?
      </Jss(Text)>
    </Jss(Card)>
    <Jss(Card)
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
          "width": "25%",
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        style={Object {}}
      >
        Something not right? 
        <Jss(Emoji)
          emoji={null}
          icon="🛠"
          label="hammer-and-wrench"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        style={Object {}}
      >
        Let us know if you're having any problems.
      </Jss(Text)>
    </Jss(Card)>
    <Jss(Card)
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
          "width": "25%",
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        style={Object {}}
      >
        Missing out? 
        <Jss(Emoji)
          emoji={null}
          icon="😭"
          label="loudly-crying-face"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        style={Object {}}
      >
        Vote for us to come to your server.
      </Jss(Text)>
    </Jss(Card)>
  </div>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      style={Object {}}
    >
      Supporting Kued
    </Jss(Text)>
  </Jss(ResponsiveModal)>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      style={Object {}}
    >
      Reporting Issues
    </Jss(Text)>
  </Jss(ResponsiveModal)>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      style={Object {}}
    >
      Server Requests
    </Jss(Text)>
  </Jss(ResponsiveModal)>
</Jss(MainLayout)>
`;
